addons/website_blog/views/website_blog_templates.xml

```diff
@@ -34,10 +34,13 @@ list of filtered posts (by date or tag).
         <t t-set="opt_blog_readable" t-value="is_view_active('website_blog.opt_blog_readable')"/>
         <t t-set="opt_blog_sidebar_show" t-value="is_view_active('website_blog.opt_blog_sidebar_show')"/>
 
-        <div id="o_wblog_blog_top">
+        <div t-att-id="blog and 'o_wblog_blog_top' or 'o_wblog_blog_top_no_blog'">
```

Comment:
Let's not change the ID, in both cases it's the same element, we might have custo relying on the ID, there is no reason to break them. Use a class :+1: